src/detect-pcre.c

```diff
@@ -274,6 +274,26 @@ static int DetectPcreSetList(int list, int set)
     return set;
 }
 
+static int DetectPcreHasUpperCase(const char *re)
+{
+    size_t len = strlen(re);
+    int is_meta = 0;
+
+    for (size_t i = 0; i < len; i++) {
```

Comment:
Honestly I've never thought of i conveying any information about its type, but instead I use it to convey info about what it is, an index ;)